### PR TITLE
Enhance list rasters and vector layer upload

### PR DESCRIPTION
**Pull request by ao-picterra**

### CODE REVIEW

File: src/picterra/client.py [hunk 1]
```diff
@@ -869,6 +873,7 @@ def upload_vector_layer(
             raster_id: The id of the raster we want to attach the vector layer to
             filename: Path to the local GeoJSOn file we want to upload
             name: Optional name to give to the vector layer
+            color: Optional color of the vector layer
```
Comment (colsil) [inline src/picterra/client.py:876]: Some details here on what format the color is in (is this a html hex code?) would be useful to document I feel

File: src/picterra/client.py [hunk 2]
```diff
@@ -878,7 +883,11 @@ def upload_vector_layer(
         upload = resp.json()
         upload_id, upload_url = upload["upload_id"], upload["upload_url"]
         _upload_file_to_blobstore(upload_url, filename)
-        data = {"name": name} if name else None
+        data = {}
```
Comment (colsil) [inline src/picterra/client.py:886]: This changes the default value of data from `None` to and empty dict `{}`, I'm guessing without testing that the end result is the same (i.e. posting no data or an empty dict to the API has the same effect?) Just wanted to check!

Comment (ao-picterra) [reply src/picterra/client.py:886]: yes, good point: just checked, and it has the same effect

File: src/picterra/client.py (hunk 1 above)
Comment (ao-picterra) [reply src/picterra/client.py]: fixed

**Review by colsil — APPROVED** [review]

**Pull request merged**